source/Core/Threads/GUIThread.cpp

```diff
@@ -996,6 +1000,40 @@ void startGUITask(void const *argument) {
   }
 #endif
 #endif
+  // Calibrate Cold Junction Compensation directly at boot, before internal components get warm.
+  uint32_t Temp = TipThermoModel::getTipInC();
+  if (!isTipDisconnected() && Temp <= 30 && getSettingValue(SettingsOptions::CalibrateCJC) > 0) {
```

Comment:
@Ralim
First of all:
Do you think this approach makes sense?

And if so, do you have time to look into this?
Especially this line:
```C#
if (!isTipDisconnected() && Temp <= 30 && getSettingValue(SettingsOptions::CalibrateCJC) > 0)
```

Reply:
Hmm okay, this makes sense.
Though you will need to have a 
```
while (preStartChecks() == 0) {
    ulTaskNotifyTake(pdTRUE, TICKS_100MS);
  }
```

Block in here to block the code progressing past this before the ADC is stable, and other tip calibration stuff is done. (See https://github.com/Ralim/IronOS/blob/434ca2647ace5bf68a4a543775cce4e9807f7f10/source/Core/Threads/PIDThread.cpp#L49 for where I copied that from)

I would yank this block of logic you have here out into a function like `performTipOffsetCalibration()`

And then just call that if the function is set.

Mostly for cleaner code paths etc.